### ISSUE TITLE
sm-ssh-ide script failure on grep during init-ssm

### ISSUE THREAD

**Issue by hidemire** [aws-samples/sagemaker-ssh-helper#62]
Hello, firstly, thank you for the wonderful library!

We encountered a problem when initializing ssm in the case when the ExecutionRole is not available for the user-profile, but only for the domain.

The `sm-ssh-ide` script, fails due to the use of grep within a pipeline combined with `set -o pipefail`.
The [grep](https://github.com/aws-samples/sagemaker-ssh-helper/blob/b6bf9ba39119784eb6ce6683ae9a6a2ed8ef4cd3/sagemaker_ssh_helper/sm-ssh-ide#L171C50-L171C70) command can exit with code 1 if "ExecutionRole" is not found, triggering the pipefail behavior and causing the script to terminate.

**Steps to Reproduce:**
1. Run the script in an environment where "ExecutionRole" is not present in the user_profile_json.
2. Observe that the script fails due to grep exiting with code 1.

**Suggested Solution:**
1. Temporarily disable `pipefail` and re-enable after the condition is processed. 

```bash
set +o pipefail
# execution_role extraction
set -o pipefail
```

2. Add `|| true;`

```bash
execution_role=$(echo "$user_profile_json" | {grep "ExecutionRole" || true;} \
      | sed -e 's/^.*"ExecutionRole": \"\([^"]*\)\".*$/\1/')
# ...
execution_role=$(echo "$domain_json" | {grep "ExecutionRole" || true;} \
        | sed -e 's/^.*"ExecutionRole": \"\([^"]*\)\".*$/\1/')
```

**Comment by ivan-khvostishkov** [comment]
Hi, @hidemire , thank you for pointing to this bug and to the solutions! I've applied the second solution to the main branch and it will appear in the next release.

**Comment by ivan-khvostishkov** [comment]
Released in v2.2.1
